### PR TITLE
[RFC] More unittests for build_stl_str_hl

### PR DESCRIPTION
**Pull request by ZviRackover**
- Improve coverage of `build_stl_str_hl`.
- Minor removal of dead code in the tested function.

### CODE REVIEW

File: src/nvim/buffer.c [hunk 1]
```diff
@@ -3273,9 +3273,6 @@ int build_stl_str_hl(
     // Two `%` in a row is the escape sequence to print a
     // single `%` in the output buffer.
     if (*fmt_p == '%') {
-      // Ignore the character if we're out of room in the output buffer.
-      if (out_p >= out_end_p)
```
Comment (ZviRackover) [inline src/nvim/buffer.c]: This was already checked in line 3260, so this code can be removed.

File: test/unit/buffer_spec.lua [hunk 2]
```diff
@@ -352,6 +364,8 @@ describe('buffer functions', function()
     statusline_test('should truncate at the first `<`', 10,
       'abc%<def%<ghijklm',   'abc<hijklm')
 
+    statusline_test('should ignore trailing %', 3, 'abc%', 'abc')
```
Comment (ZviRackover) [inline test/unit/buffer_spec.lua]: `check_stl_option` will report an error if it get this input, so not sure if we really care about this case.

**Comment by justinmk** [comment]
Nicely done, thanks.

**Comment by ZviRackover** [comment]
Sorry about that. Latest upload  should pass.

**Comment by ZviRackover** [comment]
Looks like one of the travis configurations failed due to environment issues.

**Pull request merged**